src/presentation_exchange.rs

```diff
@@ -1,80 +1,584 @@
+    pub fn name(&self) -> Option<&String> {
+        self.name.as_ref()
+    }
+
+    /// Set the purpose of the presentation definition.
+    ///
+    /// The [PresentationDefinition] MAY contain a purpose property. If present, its value MUST be a string that
+    /// describes the purpose for which the Presentation Definition's inputs are being used for.
+    pub fn set_purpose(mut self, purpose: String) -> Self {
+        self.purpose = Some(purpose);
+        self
+    }
+
+    /// Return the purpose of the presentation definition.
+    pub fn purpose(&self) -> Option<&String> {
+        self.purpose.as_ref()
+    }
+
+    /// Attach a format to the presentation definition.
+    ///
+    /// The Presentation Definition MAY include a format property. If present,
+    /// the value MUST be an object with one or more properties matching the
+    /// registered Claim Format Designations (e.g., jwt, jwt_vc, jwt_vp, etc.).
+    ///
+    /// The properties inform the [Holder](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:holder) of the Claim format configurations the [Verifier](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:verifier) can process.
+    /// The value for each claim format property MUST be an object composed as follows:
+    ///
+    /// The object MUST include a format-specific property (i.e., alg, proof_type)
+    /// that expresses which algorithms the [Verifier](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:verifier) supports for the format.
+    /// Its value MUST be an array of one or more format-specific algorithmic identifier references,
+    /// as noted in the Claim Format Designations section.
+    ///
+    /// See: [https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-definition](https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-definition)
+    pub fn set_format(mut self, format: ClaimFormat) -> Self {
+        self.format = Some(format);
+        self
+    }
+
+    /// Return the format of the presentation definition.
+    pub fn format(&self) -> Option<&ClaimFormat> {
+        self.format.as_ref()
+    }
+}
+
+/// Input Descriptors are objects used to describe the information a
+/// [Verifier](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:verifier) requires of a
+/// [Holder](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:holder).
+///
+/// All Input Descriptors MUST be satisfied, unless otherwise specified by a
+/// [Feature](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:feature).
+///
+/// See: [https://identity.foundation/presentation-exchange/spec/v2.0.0/#input-descriptor-object](https://identity.foundation/presentation-exchange/spec/v2.0.0/#input-descriptor-object)
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq, Eq)]
 pub struct InputDescriptor {
-    pub id: String,
+    id: String,
+    constraints: Constraints,
```

Comment:
Is there any reason we shouldn't add a `#[serde(default)]` attribute on `constraints` for some backwards-compatibility?

Reply:
[89db3b2](https://github.com/spruceid/oid4vp-rs/pull/9/commits/89db3b238a56dc5bd2ae43cee9cb41f9fd526817) adds this tag. This should help with backwards compatibility for schema types that don't have a `constraints` field. Although, they should, according to the spec. Might need to update the test cases to include, but at least this should resolve any compatibility-related issues. Possibly the problem with the CI.